File: FizzBuzz.py
# Simple program that takes an input n, and prints a list of numbers from 1 to n
# if a number is a multiple of 3, the number is replaced with "Fizz"
# else if a number is a multiple of 5 the number is replaced with "Buzz"
# else if a number is a multiple of both 3 and 5, the number is replaced with "FizzBuzz"
#
# Author: Ryan Cummings
# CSCI 494 Industry Methods
# 1/29/2019

# gets n from the user
def user_in():
    number = -1
    while(number <= 0):        # ensures valid input
        try:
            number = int(input("Please enter a positive integer: "))
            if number <= 0:
                print("That is not a positive integer. Please try again.")
        except ValueError:
            number = -1
            print("That is not a positive integer. Please try again.")
    return number

# Checks divisiblity and prints to console
def fizzy(n):
    for i in range(n):
        k = i + 1
        if k % 3 == 0 and k % 5 == 0:
            print("FizzBuzz")
        elif k % 3 == 0:
            print("Fizz")
        elif k % 5 == 0:
            print("Buzz")
        else:
            print(k)

def main():
    upper_bound = user_in()
    fizzy(upper_bound)

main()

File: FizzBuzz.java
/*
 * This is a function to iterate number of times the user inputs, or default to 
 * 200 iterations if input is invalid,
 * if the number is divisible by 3 fizz, 5 buzz, 7 bazz, 
 * is divisible by 3 and 5 fizzbuzz, 3 and 7, fizzbazz, 3,5,7, fizzbuzzbazz 
 */
package fizzbuzz;

/**
 *
 * @author visionaryJack/JohnWatts @date 2/12/2019
 */
import java.util.Scanner;

public class FizzBuzz {
    
    public static void main(String[] args) {
        Scanner userIn = new Scanner(System.in);
        String fizz = "Fizz", buzz = "Buzz", bazz = "Bazz"; //Strings for 3,5,7
	int iterTotal = 200;
        
        System.out.println("Please enter a integer to be iterated");
        if(userIn.hasNextInt()){ //catch exceptions, default input to 200 if non valid input
            iterTotal = userIn.nextInt();
        }else{
            System.out.println("This was not a integer, program will default to 200");
        }

        System.out.println("\nNumber entered is: "+ iterTotal +"\n");
        
	for (int i = 1; i <= iterTotal; i++)
	{
            if (i % 105 == 0) //if divisble by 105, then divisible by 3 and 5 and 7
            {   
                assert (i % 105 == 0) : "Iteration is not divisible by 105, there for is not divisible by 3 and 5 and 7.";
                System.out.println(fizz + buzz + bazz);
            }
            else if (i % 21  == 0) //if divisible by 21, then divisible by 3 and 7 
            {
                System.out.println(fizz + bazz); 
            }
            else if (i % 15  == 0) // if divisible by 15, then divisible by 3 and 5
            {
                System.out.println(fizz + buzz); 
            }
            else if (i % 3  == 0) // if divisible by 3
            {
                System.out.println(fizz);
            }
            else if (i % 5 == 0) // if divisible by 5
            {
                System.out.println(buzz);
            }
            else if (i % 7 == 0) // if divisible by 7
            {
                System.out.println(bazz);
            }
            else
            {
                System.out.println(""+i);
            }
	}
    }
}
